pkg/sdk/sdk_test.go

```diff
@@ -0,0 +1,518 @@
+					}
+					defer file.Close()
+					cType := r.Header.Get("Content-Type")
+					x := strings.Split(cType, ";")
+					if !reflect.DeepEqual(x[0], "multipart/form-data") {
+						t.Errorf("Want: %v, Got: %v", "multipart/form-data", x[0])
+					}
+					response.ToJson(w, http.StatusOK, "SUCCESS", map[string]interface{}{
+						"id": id,
+					})
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err != nil {
+					t.Errorf("Want: %v, Got: %v", nil, err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: type inference fail",
+			id:   "1",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register/{id}", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {
+					response.ToJson(w, http.StatusOK, "SUCCESS", Reader(""))
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err.Error() != "unable to parse response data" {
+					t.Errorf("Want: %v, Got: %v", "unable to parse response data", err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: json failure",
+			id:   "1",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register/{id}", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {
+
+					json.NewEncoder(w).Encode("")
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err.Error() != "json: cannot unmarshal string into Go value of type model.Response" {
+					t.Errorf("Want: %v, Got: %v", "json: cannot unmarshal string into Go value of type model.Response", err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: incorrect id in response",
+			id:   "1",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register/{id}", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {
+					response.ToJson(w, http.StatusOK, "SUCCESS", map[string]interface{}{
+						"id": "gfhgv",
+					})
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err.Error() != "incorrect id received in response" {
+					t.Errorf("Want: %v, Got: %v", "incorrect id received in response", err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: incorrect status code",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err.Error() != "non success status code received : 404" {
+					t.Errorf("Want: %v, Got: %v", "non success status code received : 404", err.Error())
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: Replace :: http post failure",
```

Comment:
Not sure where/why POST is used for this. Check.

Reply:
sir I have closed the server which is returned so when making a post request it returns an error that no connection could be made

Reply:
done